src/firmware/uapi/host/types.rs

```diff
@@ -25,6 +25,76 @@ pub enum Indeterminate<T: Debug> {
     Unknown,
 }
 
+#[derive(Debug)]
+/// Wrapper Error for Firmware or User API Errors
+pub enum UserApiError {
+    /// Firmware related errors.
+    FirmwareError(Error),
+
+    /// User API related errors.
+    ApiError(SnpCertError),
+}
+
+impl error::Error for UserApiError {
+    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
+        match self {
+            UserApiError::ApiError(uapi_error) => Some(uapi_error),
+            UserApiError::FirmwareError(firmware_error) => Some(firmware_error),
+        }
+    }
+}
+
+impl std::fmt::Display for UserApiError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let err_msg: String = match self {
+            UserApiError::FirmwareError(error) => format!("Firmware Error Encountered: {}", error),
+            UserApiError::ApiError(error) => format!("Certificate Error Encountered: {}", error),
+        };
+        write!(f, "{err_msg}")
```

Comment:
I think this should be something like

`write!(f, "{}", err_msg)`

Reply:
The way I have it should be okay. Identifiers in strings were introduced in 1.58.0 ([Announcing Rust 1.58.0 - Identifiers in Strings](https://blog.rust-lang.org/2022/01/13/Rust-1.58.0.html#captured-identifiers-in-format-strings)).

Reply:
Ah I see, my mistake. In that case, LGTM!